apps/src/weblab/WebLab.js

```diff
@@ -126,6 +114,15 @@ WebLab.prototype.init = function(config) {
             },
             {includeUserId: true}
           );
+          // The project has been reset, reload() the page now - don't resolve
+          // the promise, because that will lead to a project.save() that we
+          // don't want or need in this scenario.
+          reload();
+          reject(
```

Comment:
I considered 2 other options:
* Simply never resolving or rejecting - since we're calling `reload()`, it doesn't really matter
* Changing the "contract" so the `Promise` needs to return something (like a bool) to indicate whether or not a `save()` is needed. Again, not super important since we will already be calling `reload()`, but a little cleaner at the expense of a bigger code change